### PR TITLE
Polymer: Add the foam-storage component and ArrayBridge to support it

### PR DESCRIPTION
**Pull request by shepheb**

### CODE REVIEW

File: polymer/foam-storage.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            this.dao = EasyDAO.create({
+              model: this.model,
+              name: this.name || undefined,
+              daoType: this.type,
+              guid: true,
+              cache: true
+            }, X);
+
+            if (!this.value) this.value = [];
+            this.sink = ArrayBridge.create({
```
Comment (adamvy) [inline polymer/foam-storage.html]: Don't call it a sink.

**Comment by kgrgreer** [comment]
Technically, it's an Adapter, not a Bridge.

File: js/foam/polymer/storage/ArrayBridge.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  package: 'foam.polymer.storage',
+  name: 'ArrayBridge',
+  properties: [
+    'target',
+    'key',
+    'dao',
+    ['feedback_', 0]
+  ],
+  methods: [
+    function onChange(e) {
```
Comment (kgrgreer) [inline js/foam/polymer/storage/ArrayBridge.js]: Should this be a listener?

Comment (shepheb) [inline js/foam/polymer/storage/ArrayBridge.js]: Doesn't need to be. It was originally, and needed to be. But it's only called directly by the observer in Polymer-land; it doesn't need the binding.

**Comment by shepheb** [comment]
Renamed, and now much more robust with concurrent updates.

PTAL

File: polymer/foam-storage.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          }
+        },
+
+        observers: [
+          'onValueUpdate(value.*)'
+        ],
+
+        created: function() {
+          apar(
+            arequire('foam.dao.EasyDAO'),
```
Comment (adamvy) [inline polymer/foam-storage.html]: arequire is now a method on the context.  This will work, but it would be more correct to do X.arequire

**Pull request merged**